watertap/examples/flowsheets/case_studies/wastewater_resource_recovery/dye_desalination/dye_desalination.py

```diff
@@ -165,36 +166,52 @@ def add_costing(m):
     # create costing blocks
     dye_sep.nanofiltration.costing = UnitModelCostingBlock(**costing_kwargs)
     dye_sep.P1.costing = UnitModelCostingBlock(**costing_kwargs)
+    m.fs.zo_costing.pump_electricity.pump_cost["default"].fix(76)
 
     # aggregate unit level costs
     m.fs.zo_costing.cost_process()
 
     # create system level cost metrics
-    m.fs.brine_disposal_cost = Expression(
+    m.fs.brine_recovery_revenue = Expression(
         expr=(
             m.fs.zo_costing.utilization_factor
             * (
-                m.fs.zo_costing.waste_disposal_cost
+                m.fs.zo_costing.brine_recovery_cost
                 * pyunits.convert(
                     m.fs.permeate.properties[0].flow_vol,
                     to_units=pyunits.m**3 / m.fs.zo_costing.base_period,
                 )
             )
         ),
-        doc="Cost of disposing of saline brine/ NF permeate",
+        doc="Revenue from recovering saline brine/ NF permeate",
     )
 
-    m.fs.dye_recovery_revenue = Expression(
-        expr=(
-            m.fs.zo_costing.utilization_factor
-            * m.fs.zo_costing.dye_mass_cost
-            * pyunits.convert(
-                m.fs.dye_retentate.flow_mass_comp[0, "dye"],
-                to_units=pyunits.kg / m.fs.zo_costing.base_period,
-            )
-        ),
-        doc="Savings from dye-retentate recovered back to the plant",
-    )
+    if has_dye_revenue == True:
+        m.fs.dye_value = Expression(
+            expr=(
+                m.fs.zo_costing.utilization_factor
+                * m.fs.zo_costing.dye_recovery_cost
+                * pyunits.convert(
+                    m.fs.dye_retentate.properties[0].flow_vol,
+                    to_units=pyunits.m**3 / m.fs.zo_costing.base_period,
+                )
+            ),
+            doc="Revenue from recovering dye",
+        )
+    # Note: this is multiplied by -1 since the sign is handled implicitly
+    elif has_dye_revenue == False:
+        m.fs.dye_value = Expression(
+            expr=(
+                -1
+                * m.fs.zo_costing.utilization_factor
+                * m.fs.zo_costing.dye_disposal_cost
+                * pyunits.convert(
+                    m.fs.dye_retentate.properties[0].flow_vol,
+                    to_units=pyunits.m**3 / m.fs.zo_costing.base_period,
+                )
+            ),
+            doc="Cost of disposing of dye waste",
+        )
```

Comment:
Technically, couldn't we just have a single paramreter for dye_value (or cost) and have it go negative when we want to consider recovery value of dye and set it positive when we want to consider it as a disposal cost? Maybe we talked about this already, but at the moment, I am thinking that this can be consolidated (i.e., maybe we don't even need the Boolean arg anymore)

Reply:
@adam-a-a The reason I didn't use a single parameter here is because the dye disposal cost ($120/m3) is not equivalent to the value of recovered dye ($0.12/m3), so I think there's no way to get around having two separate expressions.

Reply:
Right, but wouldn't we only consider one case or the other? In other words, would we consider a scenario with dye disposal cost AND dye recovery value?